aQute.libg/src/aQute/lib/startlevel/StartLevelRuntimeHandler.java

```diff
@@ -198,6 +198,11 @@ public void bundleChanged(BundleEvent event) {
 							if (bundle.getBundleId() == 0)
 								return;
 
+							if (bundle.getSymbolicName() == null) {
+								logger.trace("Found bundle without a bsn %s, ignoring", bundle);
+								return;
```

Comment:
Even if the bundle does not have a BSN, shouldn't it get a startlevel? And be started?

Reply:
Nope. No bsn but it would still have a Bundle-Activator? Sounds a bizarre case that I am not interested in supporting. The only valid case is a pre 3.0 bundle when we made bsn mandatory. The code depends on BundleIdentity and I do not think it is worth the work to handle that extremely unlikely case.